### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by danipran**
(Split from #165)

Running `pytest` would work, but not `pytest -k contractor`. This PR makes various fixes to fix the flakiness of the tests and other refactoring.

Also fixes a broken warning that came up while fixing the tests.

### CODE REVIEW

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-API branch is deployed to platta: https://puistotalkoot-pr168.api.dev.hel.ninja** :rocket::rocket::rocket:

File: events/factories.py [hunk 1]
```diff
@@ -36,9 +31,15 @@ class EventFactory(factory.django.DjangoModelFactory):
...
 
     class Meta:
         model = Event
+
+    @factory.lazy_attribute
+    def location(self):
+        centroid = self.contract_zone.boundary[0].centroid
+        return Point(
+            centroid.x + randgen.uniform(0, 0.040),
+            centroid.y + +randgen.uniform(0, 0.022),
```
Comment (tuomas777) [inline events/factories.py:44]:
```suggestion
            centroid.y + randgen.uniform(0, 0.022),
```
I suppose.

**Comment by terovirtanen** [comment]
**PUISTOTALKOOT-API branch is deployed to platta: https://puistotalkoot-pr168.api.dev.hel.ninja** :rocket::rocket::rocket:

**Review by tuomas777 — APPROVED** [review]
Great improvements 💯

**Pull request merged**